### PR TITLE
DOC-1575: Fix up live demos to work with the new site

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: DOC-1575

Description of Changes:
* Reworked how the live demos were done to bring them more inline with Antora. As an example:
  * I moved the demos into the `examples` dir and made it resolve the path using the Antora content catalog APIs.
  * I removed the `site.` prefix for variables, so that it's almost identical to AsciiDoctor now except with 2 curly braces instead. This is needed as we **cannot** use AsciiDoctor subs in the demo files because it conflicts far too easily with JSDoc and JS object destructuring for example.
* Pull in the Tiny UI bundle when generating the Antora site
* Updated the generated HTML to work with the new site CSS/JS for tabs
* Fixed attributes/variables used in the demos
* Fixed the icons build script since it was using `{{site.baseurl}}` which no longer exists

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/examples/live-demos/a11ychecker/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <p>Note: This demo has been crafted to intentionally trigger most of our accessibility rules for demonstration purposes.</p>
   <hr />
   <h2><a id="example"></a>The Tiny Logo</h2>
-  <p><a href="https://www.tiny.cloud/docs/plugins/premium/a11ychecker/#liveexample" target="_blank" rel="noopener"><img src="<p><img style="display: block; margin-left: auto; margin-right: auto;" title="Tiny Logo" src="https://www.tiny.cloud/docs/images/logos/android-chrome-256x256.png" alt="TinyMCE Logo" width="128" height="128" /></p>" alt="" width="128" height="128" /></a><a href="https://www.tiny.cloud/docs/plugins/premium/a11ychecker/#liveexample" target="_blank" rel="noopener">This is a link to this demo. The same link has been added to logo to the left.</a></p>
+  <p><a href="https://www.tiny.cloud/docs/plugins/premium/a11ychecker/#liveexample" target="_blank" rel="noopener"><img src="{{logofordemos}}" alt="" width="128" height="128" /></a><a href="https://www.tiny.cloud/docs/plugins/premium/a11ychecker/#liveexample" target="_blank" rel="noopener">This is a link to this demo. The same link has been added to logo to the left.</a></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/a11ychecker/index.html:6]: We'd need to update the url slug to reflect the new docs path
ie:
https://www.tiny.cloud/docs/tinymce/6/a11ychecker.html#_interactive_example

File: modules/ROOT/examples/live-demos/annotations/index.html [hunk 2]
```diff
@@ -8,7 +8,7 @@ <h5 style="text-align: center;">Note, this includes some "enterprise/premium" fe
   <h2>Got questions or need help?</h2>
   <ul>
     <li>Our <a href="//www.tiny.cloud/docs/">documentation</a> is a great resource for learning how to configure TinyMCE.</li>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/annotations/index.html:10]: Update content links to reflect 6.0 urls
https://www.tiny.cloud/docs/tinymce/6/

File: modules/ROOT/examples/live-demos/basic-example/index.html [hunk 3]
```diff
@@ -1,12 +1,12 @@
 <textarea id="basic-example">
-  {{site.logoForDemosHTML}}
+  {{logofordemoshtml}}
   <h2 style="text-align: center;">Welcome to the TinyMCE editor demo!</h2>
 
   <h2>Got questions or need help?</h2>
 
   <ul>
     <li>Our <a href="https://www.tiny.cloud/docs/">documentation</a> is a great resource for learning how to configure TinyMCE.</li>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/basic-example/index.html:8]: find and replace these https://www.tiny.cloud/docs/
https://www.tiny.cloud/docs/tinymce/6/

File: modules/ROOT/examples/live-demos/a11ychecker/index.html (hunk 1 above)
Comment (lnewson) [reply modules/ROOT/examples/live-demos/a11ychecker/index.html:6]: This is all really out of scope for this work tbh, as this was just about getting the styling working for the demos not about updating the content. Additionally, we already have TINY-8302 to fix the demo content.

File: modules/ROOT/examples/live-demos/paste-from-word/index.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <textarea id="paste-from-word">
-  <p><img style="float: right;" title="Tiny Logo" src="https://www.tiny.cloud/docs/images/logos/android-chrome-256x256.png" alt="TinyMCE Logo" width="128" height="128" /></p>
+  {{logofordemoshtmlright}}
   <h2>TinyMCE demo: PowerPaste plugin!</h2>
 <p>PREMIUM PLUGIN: The TinyMCE <a href="https://www.tiny.cloud/docs/enterprise/paste-from-word/">PowerPaste plugin</a> automatically cleans up and transfers content &amp; images from Microsoft Word and HTML sources to ensure clean, compliant content that matches the look and feel of the site.</p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/paste-from-word/index.html:4]: closest content i could find
https://www.tiny.cloud/tinymce/6/powerpaste-support.html

File: modules/ROOT/examples/live-demos/url-conversion-absolute-1/index.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<textarea id="url-conversion-absolute-1" name="absurls">
+  <p>Some relative urls: <a href="../index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
+  <p>Some absolute urls: <a href="https://www.tiny.cloud/docs/demo/index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/url-conversion-absolute-1/index.html:3]: closest URL without an index page
https://www.tiny.cloud/tinymce/6/basic-example.html

File: modules/ROOT/examples/live-demos/url-conversion-absolute-2/index.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<textarea id="url-conversion-absolute-2" name="absurls">
+  <p>Some relative urls: <a href="../index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
+  <p>Some absolute urls: <a href="https://www.tiny.cloud/docs/demo/index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/url-conversion-absolute-2/index.html:3]: closest URL without an index page
https://www.tiny.cloud/tinymce/6/basic-example.html

File: modules/ROOT/examples/live-demos/paste-from-word/index.html (hunk 4 above)
Comment (lnewson) [reply modules/ROOT/examples/live-demos/paste-from-word/index.html:4]: Again out of scope for this task. This should be handled by TINY-8302.

File: modules/ROOT/examples/live-demos/url-conversion-none/index.html [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<textarea id="url-conversion-none">
+  <p>Some relative urls: <a href="https://www.tiny.cloud/docs/demo">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/url-conversion-none/index.html:2]: this should be a relative link as per example?

closest URL without an index page
/tinymce/6/basic-example.html

File: modules/ROOT/examples/live-demos/url-conversion-relative-1/index.html [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<textarea id="url-conversion-relative-1">
+  <p>Some relative urls: <a href="../index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/url-conversion-relative-1/index.html:2]: hmm find a relevant ../url

File: modules/ROOT/examples/live-demos/url-conversion-relative-1/index.html [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+  <p>Some absolute urls: <a href="https://www.tiny.cloud/docs/demo/index.html">content</a>. <img src="{{logofordemos}}" alt="Tiny Logo" width="128" height="128" /></p>
```
Comment (EkimChau) [inline modules/ROOT/examples/live-demos/url-conversion-relative-1/index.html:3]: closest URL without an index page
https://www.tiny.cloud/tinymce/6/basic-example.html

Comment (EkimChau) [reply modules/ROOT/examples/live-demos/url-conversion-relative-1/index.html:3]: find and replace all

**Review by EkimChau — CHANGES_REQUESTED** [review]
Requires some updates to demo example links

**Review by EkimChau — APPROVED** [review]

**Pull request merged**